cmd/fleetctl/generate_test.go

```diff
@@ -18,33 +18,27 @@ func TestGenerateMDMAppleBM(t *testing.T) {
 	require.NoError(t, err)
 	defer os.Remove(outdir)
 	publicKeyPath := filepath.Join(outdir, "public-key.crt")
-	privateKeyPath := filepath.Join(outdir, "private-key.key")
+
 	out := runAppForTest(t, []string{
 		"generate", "mdm-apple-bm",
 		"--public-key", publicKeyPath,
-		"--private-key", privateKeyPath,
 	})
 
 	require.Contains(t, out, fmt.Sprintf("Generated your public key at %s", outdir))
-	require.Contains(t, out, fmt.Sprintf("Generated your private key at %s", outdir))
 
-	// validate that the keypair is valid
-	cert, err := tls.LoadX509KeyPair(publicKeyPath, privateKeyPath)
+	// validate that the certificate is valid
+	certPEMBlock, err := os.ReadFile(publicKeyPath)
 	require.NoError(t, err)
 
-	parsed, err := x509.ParseCertificate(cert.Certificate[0])
+	parsed, err := x509.ParseCertificate(certPEMBlock)
 	require.NoError(t, err)
 	require.Equal(t, "FleetDM", parsed.Issuer.CommonName)
 }
 
 func TestGenerateMDMApple(t *testing.T) {
-	t.Run("missing input", func(t *testing.T) {
-		runAppCheckErr(t, []string{"generate", "mdm-apple"}, `Required flags "email, org" not set`)
-		runAppCheckErr(t, []string{"generate", "mdm-apple", "--email", "user@example.com"}, `Required flag "org" not set`)
-		runAppCheckErr(t, []string{"generate", "mdm-apple", "--org", "Acme"}, `Required flag "email" not set`)
-	})
-
 	t.Run("CSR API call fails", func(t *testing.T) {
+		// TODO(roberto): update when the new endpoint to get a CSR is ready
+		t.Skip()
```

Comment:
calling this and the other skip below, we should remove this once we have the endpoints. I have manually tested tho.